lib/trusted-mpt-light-client-types/src/signed_data.rs

```diff
@@ -0,0 +1,30 @@
+use unionlabs::{
+    encoding::{Bincode, Encode},
+    primitives::{H256, H512},
+};
+
+#[derive(Debug, Clone, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "bincode", derive(bincode::Encode, bincode::Decode))]
+pub struct SignedData<T: Encode<Bincode>> {
+    data: T,
+    signature: H512,
+}
+
+impl<T: Encode<Bincode> + Clone> SignedData<T> {
+    pub fn new(data: T, signature: H512) -> Self {
+        SignedData { data, signature }
+    }
+
+    pub fn unwrap_verified<V: Fn(&[u8], H512, H256) -> Option<bool>>(
```

Comment:
put the Encoding param on this function (E: Encoding, and then add where T: Encode<E>)